network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/ActivePowerLimitsAdderImpl.java

```diff
@@ -0,0 +1,122 @@
+package com.powsybl.network.store.iidm.impl;
+
+import com.powsybl.iidm.network.*;
+import com.powsybl.network.store.model.LimitsAttributes;
+import com.powsybl.network.store.model.TemporaryCurrentLimitAttributes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Comparator;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+public class ActivePowerLimitsAdderImpl<S, OWNER extends LimitsOwner<S>> implements ActivePowerLimitsAdder {
+
+    private static final Comparator<Integer> ACCEPTABLE_DURATION_COMPARATOR = (acceptableDuraction1, acceptableDuraction2) -> acceptableDuraction2 - acceptableDuraction1;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActivePowerLimitsAdderImpl.class);
+
+    private final OWNER owner;
+
+    private final S side;
+
+    private double permanentLimit = Double.NaN;
+
+    private TreeMap<Integer, TemporaryCurrentLimitAttributes> temporaryLimits = new TreeMap<>(ACCEPTABLE_DURATION_COMPARATOR);
+
+    ActivePowerLimitsAdderImpl(S side, OWNER owner) {
+        this.owner = owner;
+        this.side = side;
+    }
+
+    @Override
+    public ActivePowerLimitsAdder setPermanentLimit(double permanentLimit) {
+        this.permanentLimit = permanentLimit;
+        return this;
+    }
+
+    public OWNER getOwner() {
+        return owner;
+    }
+
+    @Override
+    public LoadingLimitsAdder.TemporaryLimitAdder beginTemporaryLimit() {
+        return new TemporaryLimitActiveLimitAdderImpl(this);
+    }
+
+    @Override
+    public double getPermanentLimit() {
+        return this.permanentLimit;
+    }
+
+    @Override
+    public double getTemporaryLimitValue(int acceptableDuration) {
+        TemporaryCurrentLimitAttributes tl = getTemporaryLimits().get(acceptableDuration);
+        return tl != null ? tl.getValue() : Double.NaN;
+
+    }
+
+    @Override
+    public boolean hasTemporaryLimits() {
+        return !temporaryLimits.isEmpty();
+    }
+
+    public void addTemporaryLimit(TemporaryCurrentLimitAttributes temporaryLimitAttribute) {
+        temporaryLimits.put(temporaryLimitAttribute.getAcceptableDuration(), temporaryLimitAttribute);
+    }
+
+    public SortedMap<Integer, TemporaryCurrentLimitAttributes> getTemporaryLimits() {
+        return temporaryLimits;
+    }
+
+    private void checkTemporaryLimits() {
+        // check temporary limits are ok and are consistents with permanent
```

Comment:
is this copy pasted from powsybl-core ? Can we avoid it (maybe in another PR ?)

Reply:
It's from powsybl-core indeed but why should I remove it, it makes sense to have the same check no ?